vowpalwabbit/io_buf.h

```diff
@@ -361,6 +363,30 @@ inline size_t bin_text_read_write_fixed_validated(
   return nbytes;
 }
 
+namespace VW
+{
+template <typename T>
+size_t process_model_field(io_buf& io, T& var, bool read, VW::string_view name, bool text)
+{
+  // It is not valid to read a text based field.
+  assert(!(read && text));
+
+  auto* data = reinterpret_cast<char*>(&var);
```

Comment:
we are assuming anything `T` can be cast into a char representation right?

Reply:
Yes, we are. We can add some template checks to make sure it is a type where this makes sense. POD is probably what we want to check for. This being a template allows us to overload for types that can't be trivially bit cast to bytes (like std::string)